pkg/byteorder/byteorder.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package byteorder
+
+import (
+	"encoding/binary"
+	"fmt"
+	"reflect"
+
+	"github.com/vishvananda/netlink/nl"
+)
+
+// Native is set to the host ByteOrder type. This should normally be either
+// BigEndian or LittleEndian.
+var Native = nl.NativeEndian()
+
+// reverse returns a reversed slice of b.
+func reverse(b []byte) []byte {
+	size := len(b)
+	c := make([]byte, size)
+
+	for i, j := size-1, 0; i >= 0; i, j = i-1, j+1 {
+		c[j] = b[i]
+	}
+
+	return c
+}
+
+// HostToNetwork converts b to the networking byte order.
+func HostToNetwork(b interface{}) interface{} {
+	switch b.(type) {
+	case uint16:
+		return nl.Swap16(b.(uint16))
+	case uint32:
+		return nl.Swap32(b.(uint32))
+	default:
+		panic(unsupported(b))
+	}
+}
+
+// NetworkToHost converts n to host byte order.
+func NetworkToHost(n interface{}) interface{} {
+	switch n.(type) {
+	case uint16:
+		return Native.Uint16(nl.Uint16Attr(n.(uint16)))
+	case uint32:
+		return Native.Uint32(nl.Uint32Attr(n.(uint32)))
```

Comment:
Not sure. Will fix and push out later with some other queued things.
Just need to wait for the local tests first. Thanks.